Snowplow/SPEmitter.m

```diff
@@ -172,27 +202,81 @@ - (void) sendEvents {
+                    
+                    // Reset the STM as it is going individually
+                    [eventPayload setValue:[NSString stringWithFormat:@"%ld", (long)[SPUtilities getTimestamp]] forKey:kSPSentTimestamp];
+                    
+                    // Build and Send the event!
+                    [singleEventArray addObject:eventPayload];
+                    [singleIndexArray addObject:[listValues[j] objectForKey:@"ID"]];
+                    SPSelfDescribingJson *payload = [[SPSelfDescribingJson alloc] initWithSchema:kSPPayloadDataSchema
+                                                                                         andData:singleEventArray];
+                    [self sendEventWithRequest:[self getRequestPostWithData:payload] andIndex:singleIndexArray andResultArray:sendResults andOversize:YES];
+                } else if (totalByteSize + payloadByteSize > _byteLimitPost) {
+                    // Adding this event to the accumulated array would exceed the limit.
+                    SPSelfDescribingJson *payload = [[SPSelfDescribingJson alloc] initWithSchema:kSPPayloadDataSchema
+                                                                                         andData:eventArray];
+                    [self sendEventWithRequest:[self getRequestPostWithData:payload] andIndex:indexArray andResultArray:sendResults andOversize:NO];
+                    
+                    // Reset collections and STM
+                    eventArray = [[NSMutableArray alloc] init];
+                    indexArray = [[NSMutableArray alloc] init];
+                    stm = [SPUtilities getTimestamp];
+                    
+                    // Add new STM to the payload
+                    [eventPayload setValue:[NSString stringWithFormat:@"%ld", (long)stm] forKey:kSPSentTimestamp];
+                    
+                    // Add event to collections
+                    [eventArray addObject:eventPayload];
+                    [indexArray addObject:[listValues[j] objectForKey:@"ID"]];
+                    
+                    // Update byte count
+                    totalByteSize = payloadByteSize;
+                } else {
+                    // Add event to collections
+                    [eventArray addObject:eventPayload];
+                    [indexArray addObject:[listValues[j] objectForKey:@"ID"]];
+                    
+                    // Update byte count
+                    totalByteSize += payloadByteSize;
+                }
             }
             
-            NSMutableDictionary *payload = [[NSMutableDictionary alloc] init];
-            [payload setValue:kSPPayloadDataSchema forKey:@"schema"];
-            [payload setValue:eventArray forKey:@"data"];
-            [self sendSyncRequest:[self getRequestPostWithData:payload] withIndex:indexArray withResultPointer:sendResults];
+            // If we have not sent all of the events...
+            if (eventArray.count > 0) {
```

Comment:
What do you think of simply adding 21 bytes to the payloadByteSize instead of early adding the `stm` to ensure correct order of sent events?

21 bytes gotten from: `"stm":"1443019220000"`